src/admin/index.js

```diff
@@ -990,6 +994,86 @@ module.exports = ({
     })
   }
 
+  /**
+   * Delete topic records up to the selected partition offsets
+   *
+   * @param {string} topic
+   * @param {Array<SeekEntry>} partitions
+   * @return {Promise}
+   *
+   * @typedef {Object} SeekEntry
+   * @property {number} partition
+   * @property {string} offset
+   */
+  const deleteTopicRecords = async ({ topic, partitions }) => {
+    if (!topic || typeof topic !== 'string') {
+      throw new KafkaJSNonRetriableError(`Invalid topic ${topic}`)
+    }
+
+    if (!partitions || partitions.length === 0) {
+      throw new KafkaJSNonRetriableError(`Invalid partitions`)
+    }
+
+    const partitionsByBroker = cluster.findLeaderForPartitions(
+      topic,
+      partitions.map(p => p.partition)
+    )
+
+    const partitionsFound = flatten(values(partitionsByBroker))
+    const topicOffsets = await fetchTopicOffsets(topic)
+
+    partitions.forEach(({ partition, offset }) => {
+      // warn if no broker/leader found for partition
+      if (!partitionsFound.includes(partition)) {
```

Comment:
I think I disagree with this choice, as it means that without human intervention you would never know that the operation failed. If we fail to find a leader for the partition, I think it's preferable that the operation fails rather than partially completing without knowing what partitions actually failed. Otherwise, how do you react to this in a programmatic way?

Reply:
tried to re-use the new `KafkaJSDeleteTopicRecordsError` structure, as a way to send back an error showing all cases of "leader not found" in the event it's more than 1. Let me know what you think (makes the code a bit less readable because it's collecting the errors in an array, but probably better for the user?)

Reply:
I agree 👍